scripts/globals/abilities/shadowbind.lua

```diff
@@ -14,15 +14,34 @@ require("scripts/globals/status");
 -----------------------------------
 
 function onAbilityCheck(player,target,ability)
-    return 0,0;
+    if ((player:getWeaponSkillType(SLOT_RANGED) == SKILL_MRK and player:getWeaponSkillType(SLOT_AMMO) == SKILL_MRK) or
+    (player:getWeaponSkillType(SLOT_RANGED) == SKILL_ARC and player:getWeaponSkillType(SLOT_AMMO) == SKILL_ARC)) then
+        return 0,0;
+    end
+    return 216,0; -- You do not have an appropriate ranged weapon equipped.
 end;
 
 -----------------------------------
 -- onUseAbility
 -----------------------------------
 
-function onUseAbility(player,target,ability)
-    if (player:getWeaponSkillType(SLOT_RANGED) == SKILL_MRK) then
+function onUseAbility(player,target,ability,action)
+
+    if (player:getWeaponSkillType(SLOT_RANGED) == SKILL_MRK) then -- can't have your crossbow/gun held like a bow, now can we?
         action:animation(target:getID(), action:animation(target:getID()) + 1);
     end
-end;
+
+    local duration = 30 + player:getMod(MOD_SHADOW_BIND_EXT);
+
+     -- TODO: Acc penalty for /RNG, acc vs. mob level?
+    if (math.random(0, 100) >= target:getMod(MOD_BINDRES) and target:hasStatusEffect(EFFECT_BIND) == false) then
+        target:addStatusEffect(EFFECT_BIND, 0, 0, duration);
+        ability:setMsg(277); -- Target is bound.
+    else
+        ability:setMsg(158); -- Player uses Shadowbind, but misses.
+    end
+
+    player:removeAmmo(); -- Shadowbind depletes one round of ammo.
```

Comment:
Wiki doesn't mention anything about either of those abilities... hell bgwiki doesn't even mention anything about it consuming ammo.